src/builder.ts

```diff
@@ -335,6 +346,13 @@ export class SchemaBuilder {
           }
           const fieldName = resolvedConfig.alias || field.name
           const type = resolvedConfig.type || field.outputType.type
+
+          if (
+            !this.assertOutputTypeIsDefined(typeName, fieldName, type, stage)
```

Comment:
Can we use naming to clarify the difference between `typeName` and `type`. e.g. `modelType`? What is it?

Reply:
yeah we can! it makes a bit more sense when you see where `type` is going (in the `type` property of `t.field`). Here `type` is the OutputType, and `typeName`, the parent typename in which the dynamic is called. I'll find better names!